### PR TITLE
Fixes #13657: Script rudder-support-info does not mask credentials

### PR DESCRIPTION
**Pull request by ncharles**
https://issues.rudder.io/issues/13657

### CODE REVIEW

File: rudder-agent/SOURCES/debug-script/lib/collect [hunk 1]
```diff
@@ -105,6 +105,29 @@ collect_files() {
     cp ${RUDDER_OPT}/etc/rudder-networks* "${TARGET_DIRECTORY}/rudder/" 2>/dev/null
     cp ${RUDDER_VAR}/ncf/local/ncf.conf "${TARGET_DIRECTORY}/rudder/" 2>/dev/null
 
+
+    # Hash the passwords
+    # This is ad-hoc, as most files/keys differ
```
Comment (peckpeck) [inline rudder-agent/SOURCES/debug-script/lib/collect]: wouldn't be easier to read if you has a hash_secret() function ?

Comment (ncharles) [reply rudder-agent/SOURCES/debug-script/lib/collect]: i would pass it the whole regex and the filename ?

Comment (peckpeck) [reply rudder-agent/SOURCES/debug-script/lib/collect]: wouldn't the key name be sufficient ?

File: rudder-agent/SOURCES/debug-script/lib/collect [hunk 2]
```diff
@@ -105,6 +105,29 @@ collect_files() {
...
+    perl -MDigest::SHA=sha512_hex -pe 's/^(ldap.authpw=)(.*)$/"$1".sha512_hex$2/e' "${TARGET_DIRECTORY}/rudder/inventory-web.properties" > "${TARGET_DIRECTORY}/rudder/inventory-web.properties.hashed"
+    mv "${TARGET_DIRECTORY}/rudder/inventory-web.properties.hashed" "${TARGET_DIRECTORY}/rudder/inventory-web.properties"
+    perl -MDigest::SHA=sha512_hex -pe 's/^(ldap.authpw=)(.*)$/"$1".sha512_hex$2/e' "${TARGET_DIRECTORY}/rudder/rudder-web.properties" > "${TARGET_DIRECTORY}/rudder/rudder-web.properties.hashed"
+    mv "${TARGET_DIRECTORY}/rudder/rudder-web.properties.hashed" "${TARGET_DIRECTORY}/rudder/rudder-web.properties"
+    perl -MDigest::SHA=sha512_hex -pe 's/^(rudder.jdbc.password=)(.*)$/"$1".sha512_hex$2/e' "${TARGET_DIRECTORY}/rudder/rudder-web.properties" > "${TARGET_DIRECTORY}/rudder/rudder-web.properties.hashed"
+    mv "${TARGET_DIRECTORY}/rudder/rudder-web.properties.hashed" "${TARGET_DIRECTORY}/rudder/rudder-web.properties"
+
+    perl -MDigest::SHA=sha512_hex -pe 's/^(rudder.webdav.password==)(.*)$/"$1".sha512_hex$2/e' "${TARGET_DIRECTORY}/rudder/rudder-web.properties" > "${TARGET_DIRECTORY}/rudder/rudder-web.properties.hashed"
```
Comment (peckpeck) [inline rudder-agent/SOURCES/debug-script/lib/collect]: double == ?

**Comment by ncharles** [comment]
Commit modified

**Comment by ncharles** [comment]
Commit modified

**Pull request merged**